blockchain/v2/scheduler.go

```diff
@@ -608,35 +570,45 @@ func (sc *scheduler) handleBlockProcessError(event pcBlockVerificationFailure) (
 	}
 
 	if sc.allBlocksProcessed() {
-		return scFinishedEv{}, nil
+		return scFinishedEv{reason: "error on last block"}, nil
 	}
 
 	return noOp, nil
 }
 
-func (sc *scheduler) handleAddNewPeer(event addNewPeer) (Event, error) {
+func (sc *scheduler) handleAddNewPeer(event bcAddNewPeer) (Event, error) {
 	err := sc.addPeer(event.peerID)
 	if err != nil {
 		return scSchedulerFail{reason: err}, nil
```

Comment:
So it seems these failure events don't trigger anything right? Note some of them seem like genuine internal errors (ie. the ones in `handleTrySchedule`), while others seem highly probable in practice. This one especially because the peer layer may try to re-add a peer we disconnected from in the past. Other fails are due to race conditions. So it certainly doesn't seem like we can consider these fatal without more work, and we should double check that its ok that these will happen.

Reply:
yeah, we need to comb through the cases and make sure we are not in a corrupt state.

Reply:
Made a follow up issue. #4415